lean4/src/putnam_1989_b6.lean

```diff
@@ -2,18 +2,31 @@ import Mathlib
 
 open Nat Filter Topology Set ProbabilityTheory
 
+
 -- Note: uses (ℝ → ℝ) instead of (Set.Icc 0 1 → ℝ)
 /--
 Let $(x_1,x_2,\dots,x_n)$ be a point chosen at random from the $n$-dimensional region defined by $0<x_1<x_2<\dots<x_n<1$. Let $f$ be a continuous function on $[0,1]$ with $f(1)=0$. Set $x_0=0$ and $x_{n+1}=1$. Show that the expected value of the Riemann sum $\sum_{i=0}^n (x_{i+1}-x_i)f(x_{i+1})$ is $\int_0^1 f(t)P(t)\,dt$, where $P$ is a polynomial of degree $n$, independent of $f$, with $0 \leq P(t) \leq 1$ for $0 \leq t \leq 1$.
 -/
 theorem putnam_1989_b6
     (n : ℕ) [NeZero n]
     (I : (Fin n → ℝ) → Fin (n + 2) → ℝ)
-    (I_def : ∀ x i, I x i = if i = 0 then 0 else if i = - 1 then 1 else x (i : ℕ).pred)
+    (I_def : ∀ x i, I x i = by
```

Comment:
The fix here is `open scoped Fin.CommRing in`

Reply:
Or to use `Fin.last` for `-1`

Reply:
done